diff -ur '--exclude=.git' OrangePi_Build/OrangePiRK3399/scripts/00_rootfs_build.sh OrangePiRK3399/scripts/00_rootfs_build.sh
--- OrangePi_Build/OrangePiRK3399/scripts/00_rootfs_build.sh	2019-10-07 11:40:35.514759688 +0800
+++ OrangePiRK3399/scripts/00_rootfs_build.sh	2019-10-07 10:24:20.751793448 +0800
@@ -12,7 +12,7 @@
 fi
 
 if [ -z $1 ]; then
-	DISTRO="xenial"
+	DISTRO="bionic"
 else
 	DISTRO=$1
 fi
@@ -61,7 +61,7 @@
 fi
 
 if [ -z "$DISTRO" ]; then
-	DISTRO="xenial"
+	DISTRO="bionic"
 fi
 
 TEMP=$(mktemp -d)
@@ -83,8 +83,8 @@
 METHOD="download"
 
 case $DISTRO in
-	xenial)
-		ROOTFS="http://cdimage.ubuntu.com/ubuntu-base/releases/16.04/release/ubuntu-base-16.04.5-base-arm64.tar.gz"
+	bionic)
+		ROOTFS="http://cdimage.ubuntu.com/ubuntu-base/releases/18.04/release/ubuntu-base-18.04.3-base-arm64.tar.gz"
 		;;
 	jessie|stretch)
                 ROOTFS="${DISTRO}-base-arm64.tar.gz"
@@ -109,7 +109,7 @@
         EXCLUDE="--exclude=init,systemd-sysv"
         EXTR="--keep-debootstrap-dir"
         RELEASE=jessie
-        APT_SERVER=mirrors.ustc.edu.cn
+        APT_SERVER=mirrors.mit.edu
         APT_INCLUDES="--include=apt-transport-https,apt-utils,ca-certificates,debian-archive-keyring,dialog,sudo,systemd,sysvinit-utils,parted,dbus,openssh-server,alsa-utils,rng-tools,locales"
         QEMU_BINARY="/usr/bin/qemu-aarch64-static"
 
@@ -132,13 +132,8 @@
         # Complete the bootstrapping process
         echo -e "\e[1;31m Start debootstrap second stage.....\e[0m"
 
-        chroot rootfs mount -t proc proc /proc || true
-        chroot rootfs mount -t sysfs sys /sys || true
-        chroot rootfs /debootstrap/debootstrap --second-stage
-        chroot rootfs umount /sys || true
-        chroot rootfs umount /proc || true
 
-        #do_chroot /debootstrap/debootstrap --second-stage
+        do_chroot /debootstrap/debootstrap --second-stage
 
         # keeping things clean as this is copied later again
         rm -f rootfs/usr/bin/qemu-aarch64-static
@@ -180,11 +175,20 @@
 
 do_chroot() {
 	cmd="$@"
-	chroot "$DEST" mount -t proc proc /proc || true
-	chroot "$DEST" mount -t sysfs sys /sys || true
+	#chroot "$DEST" mount -t proc /proc proc || true
+	#chroot "$DEST" mount -t sysfs sys /sys || true
+	mount -t proc /proc $DEST/proc || true
+	mount -o bind /sys $DEST/sys || true
+	mount -o bind /dev $DEST/dev || true
+	mount -o bind /dev/pts $DEST/dev/pts || true
+
 	chroot "$DEST" $cmd
-	chroot "$DEST" umount /sys
-	chroot "$DEST" umount /proc
+	exit
+	umount -l $DEST/dev/pts || true
+	umount -l $DEST/dev || true
+	umount -l $DEST/proc || true
+	umount -l $DEST/sys || true
+	
 }
 
 add_platform_scripts() {
@@ -275,31 +279,31 @@
 	local release="$1"
 	local aptsrcfile="$DEST/etc/apt/sources.list"
 	cat > "$aptsrcfile" <<EOF
-deb https://mirrors.ustc.edu.cn/debian/ ${release} main contrib non-free
-#deb-src https://mirrors.ustc.edu.cn/debian/ ${release} main contrib non-free
+deb https://mirrors.mit.edu/debian/ ${release} main contrib non-free
+#deb-src https://mirrors.mit.edu/debian/ ${release} main contrib non-free
 
-deb https://mirrors.ustc.edu.cn/debian/ ${release}-updates main contrib non-free
-#deb-src https://mirrors.ustc.edu.cn/debian/ ${release}-updates main contrib non-free
+deb https://mirrors.mit.edu/debian/ ${release}-updates main contrib non-free
+#deb-src https://mirrors.mit.edu/debian/ ${release}-updates main contrib non-free
 
-deb https://mirrors.ustc.edu.cn/debian/ ${release}-backports main contrib non-free
-#deb-src https://mirrors.ustc.edu.cn/debian/ ${release}-backports main contrib non-free
+deb https://mirrors.mit.edu/debian/ ${release}-backports main contrib non-free
+#deb-src https://mirrors.mit.edu/debian/ ${release}-backports main contrib non-free
 EOF
 }
 
 add_ubuntu_apt_sources() {
 	local release="$1"
 	cat > "$DEST/etc/apt/sources.list" <<EOF
-deb http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial main restricted universe multiverse
-#deb-src https://mirrors.ustc.edu.cn/ubuntu-ports/ xenial main main restricted universe multiverse
-deb http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-updates main restricted universe multiverse
-#deb-src https://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-updates main restricted universe multiverse
-deb http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-backports main restricted universe multiverse
-#deb-src https://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-backports main restricted universe multiverse
-deb http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-security main restricted universe multiverse
-#deb-src https://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-security main restricted universe multiverse
+deb http://mirrors.mit.edu/ubuntu-ports/ bionic main restricted universe multiverse
+#deb-src https://mirrors.mit.edu/ubuntu-ports/ bionic main main restricted universe multiverse
+deb http://mirrors.mit.edu/ubuntu-ports/ bionic-updates main restricted universe multiverse
+#deb-src https://mirrors.mit.edu/ubuntu-ports/ bionic-updates main restricted universe multiverse
+deb http://mirrors.mit.edu/ubuntu-ports/ bionic-backports main restricted universe multiverse
+#deb-src https://mirrors.mit.edu/ubuntu-ports/ bionic-backports main restricted universe multiverse
+deb http://mirrors.mit.edu/ubuntu-ports/ bionic-security main restricted universe multiverse
+#deb-src https://mirrors.mit.edu/ubuntu-ports/ bionic-security main restricted universe multiverse
 
-#deb http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-proposed main restricted universe multiverse
-#deb-src http://mirrors.ustc.edu.cn/ubuntu-ports/ xenial-proposed main restricted universe multiverse
+#deb http://mirrors.mit.edu/ubuntu-ports/ bionic-proposed main restricted universe multiverse
+#deb-src http://mirrors.mit.edu/ubuntu-ports/ bionic-proposed main restricted universe multiverse
 EOF
 }
 
@@ -311,10 +315,10 @@
 # Run stuff in new system.
 case $DISTRO in
 
-	xenial|jessie|stretch)
+	bionic|jessie|stretch)
 		rm "$DEST/etc/resolv.conf"
 		cp /etc/resolv.conf "$DEST/etc/resolv.conf"
-		if [ "$DISTRO" = "xenial" ]; then
+		if [ "$DISTRO" = "bionic" ]; then
 			DEB=ubuntu
 			DEBUSER=orangepi
 			EXTRADEBS="software-properties-common ubuntu-minimal"
@@ -336,13 +340,19 @@
 		cat > "$DEST/second-phase" <<EOF
 #!/bin/sh
 export DEBIAN_FRONTEND=noninteractive
-locale-gen en_US.UTF-8
+export LC_ALL=en_US.UTF-8
+export LANGUAGE=en_US:en
+export LANG=en_US.UTF-8
 apt-get -y update
-apt-get -y install dosfstools curl xz-utils iw rfkill wpasupplicant usbutils openssh-server alsa-utils $EXTRADEBS
+apt-get -y install sudo locales dialog
+locale-gen en_US.UTF-8
+
+apt-get -y install dosfstools curl xz-utils iw rfkill wpasupplicant usbutils openssh-server alsa-utils
+apt-get -y install $EXTRADEBS
 apt-get -y install rsync u-boot-tools vim parted network-manager usbmount git autoconf gcc libtool libsysfs-dev pkg-config libdrm-dev xutils-dev hostapd dnsmasq
 apt-get -y remove --purge ureadahead
 $ADDPPACMD
-apt-get -y update
+#apt-get -y update
 $DISPTOOLCMD
 adduser --gecos $DEBUSER --disabled-login $DEBUSER --uid 1000
 adduser --gecos root --disabled-login root --uid 0
@@ -360,11 +370,12 @@
 		chmod +x "$DEST/second-phase"
 		do_chroot /second-phase
 
-if [ $TYPE = "1" -a $DISTRO="xenial" ]; then
+if [ $TYPE = "1" -a $DISTRO="bionic" ]; then
                 cat > "$DEST/type-phase" <<EOF
 #!/bin/sh
 apt-get -y install xubuntu-desktop vlc
-apt remove snapd
+apt-get -y remove snapd
+apt-get -y dist-upgrade 
 apt-get -y autoremove
 apt-get clean
 EOF
@@ -383,6 +394,7 @@
                 do_chroot /type-phase
 fi
 
+		mkdir -p "$DEST/etc/network/interfaces.d/"
 		cat > "$DEST/etc/network/interfaces.d/eth0" <<EOF
 auto eth0
 iface eth0 inet dhcp
Only in OrangePi_Build/OrangePiRK3399/scripts/: 00_rootfs_build.sh.orig
diff -ur '--exclude=.git' OrangePi_Build/OrangePiRK3399/scripts/01_rootfs_build.sh OrangePiRK3399/scripts/01_rootfs_build.sh
--- OrangePi_Build/OrangePiRK3399/scripts/01_rootfs_build.sh	2019-10-07 11:40:35.514759688 +0800
+++ OrangePiRK3399/scripts/01_rootfs_build.sh	2019-10-07 04:20:48.186998508 +0800
@@ -10,7 +10,7 @@
 fi
 
 if [ -z $1 ]; then
-	DISTRO="xenial"
+	DISTRO="bionic"
 else
 	DISTRO=$1
 fi
Only in OrangePi_Build/OrangePiRK3399/scripts/: 01_rootfs_build.sh.orig
diff -ur '--exclude=.git' OrangePi_Build/OrangePiRK3399/scripts/build_image.sh OrangePiRK3399/scripts/build_image.sh
--- OrangePi_Build/OrangePiRK3399/scripts/build_image.sh	2019-10-07 11:40:35.514759688 +0800
+++ OrangePiRK3399/scripts/build_image.sh	2019-10-07 10:27:36.540002227 +0800
@@ -22,7 +22,7 @@
 	PLATFORM=$2
 fi
 
-if [ $3 = "1" ]; then
+if [ ! -z $3 ] && [ $3 = "1" ]; then
 	IMAGETYPE="desktop"
 	disk_size="3800"
 else
@@ -30,6 +30,13 @@
 	disk_size="1200"
 fi
 
+if [ -z $4 ]; then
+	VAR=0
+else
+	VAR=$4
+fi
+
+
 
 BUILD="$ROOT/external"
 TOOLPATH=${BUILD}/rkbin/tools
@@ -38,7 +45,6 @@
 IMAGENAME="OrangePi_${PLATFORM}_${DISTRO}_${IMAGETYPE}_${VER}.img"
 IMAGE="$OUTPUT/images/$IMAGENAME"
 ROOTFS="$OUTPUT/rootfs"
-VAR=$4
 PATH=$PATH:$TOOLPATH
 source $ROOT/scripts/partitions.sh
 
@@ -86,12 +92,22 @@
 
 # Create additional ext4 file system for rootfs
 rootfs_size=$((disk_size-128))  # $disk_size - $ROOTFS_START*512 / 1024 / 1024
+echo "____________________________________"
+echo "rootfs_size=$rootfs_size"
+
 dd if=/dev/zero bs=1M count=$rootfs_size of=${IMAGE}2
+echo "____________________________________"
+echo "starting mkfs.ext4"
 mkfs.ext4 -F -b 4096 -E stride=2,stripe-width=1024 -L rootfs ${IMAGE}2
 
 if [ ! -d /media/tmp ]; then
     mkdir -p /media/tmp
 fi
+echo "____________________________________"
+echo "mount img2"
+echo "ROOT=$ROOT"
+echo "OUTPUT=$OUTPUT"
+echo "VAR=$VAR"
 
 mount -t ext4 ${IMAGE}2 /media/tmp
 
Only in OrangePi_Build/OrangePiRK3399/scripts/: build_image.sh.orig
Only in OrangePi_Build/OrangePiRK3399/scripts/: Build_OrangePi.sh
diff -ur '--exclude=.git' OrangePi_Build/OrangePiRK3399/scripts/build.sh OrangePiRK3399/scripts/build.sh
--- OrangePi_Build/OrangePiRK3399/scripts/build.sh	2019-10-07 11:40:35.514759688 +0800
+++ OrangePiRK3399/scripts/build.sh	2019-10-07 04:18:04.295637325 +0800
@@ -192,7 +192,7 @@
 	MENUSTR="Distro Options"
 	OPTION=$(whiptail --title "OrangePi Build System" \
 		--menu "$MENUSTR" 20 60 3 --cancel-button Finish --ok-button Select \
-		"0"   "Ubuntu Xenial" \
+		"0"   "Ubuntu Bioic" \
 		"1"   "Debian Jessie" \
 		3>&1 1>&2 2>&3)
 
@@ -223,7 +223,7 @@
 	fi
 
 	if [ $OPTION = "0" ]; then
-		TMP_DISTRO="xenial"
+		TMP_DISTRO="bionic"
         elif [ $OPTION = "1" ]; then
                 TMP_DISTRO="jessie"
         fi
@@ -365,3 +365,11 @@
 		--msgbox "Pls select correct option" 10 50 0
 	exit 0
 fi
+
+mnts="dev/pts dev proc sys"
+for mmnt in $mnts; do 
+    cmd="sudo umount -l output/rootfs/$mmnt"
+    #echo $cmd
+    $cmd
+done 
+
Only in OrangePi_Build/OrangePiRK3399/scripts/: build.sh.orig
Only in OrangePi_Build/OrangePiRK3399/scripts/: README.md
Only in OrangePiRK3399/scripts/: umounts.sh
